src/tyr/route_serializer_osrm.cc

```diff
@@ -325,6 +334,37 @@ struct IntersectionEdges {
   }
 };
 
+// Forward declaration
+valhalla::baldr::json::MapPtr serialize_incident(const TripLeg_Node_Incident& incident);
+
+// Serializes incidents and adds to json-document
+void addsIncidents(
+    const google::protobuf::RepeatedPtrField<valhalla::TripLeg_Node_Incident>& incidents,
+    json::Jmap& doc) {
+  if (incidents.size() == 0) {
+    // No incidents, nothing to do
+    return;
+  }
+  json::ArrayPtr serialized_incidents = std::shared_ptr<json::Jarray>(new json::Jarray());
+  for (const auto& incident : incidents) {
+    {
+      // Bring up any already existing array
+      auto existing = doc.find("incidents");
+      if (existing != doc.end()) {
+        if (auto* ptr =
+                boost::get<std::shared_ptr<valhalla::baldr::json::Jarray>>(&existing->second)) {
+          serialized_incidents = *ptr;
+        } else {
+          throw std::logic_error("This should not be possible");
```

Comment:
Can we put a bit more information into the error message? Like "could not get existing incidents" or some such.